### PR TITLE
Use BOOST_OVERRIDE

### PR DESCRIPTION
**Pull request by EugeneZelenko**
Use BOOST_OVERRIDE to fix GCC -Wsuggest-override and Clang-tidy modernize-use-override warnings.
Alphabetical order of STL headers.

### CODE REVIEW

File: include/boost/numeric/conversion/converter_policies.hpp [hunk 1]
```diff
@@ -10,14 +10,13 @@
 #ifndef BOOST_NUMERIC_CONVERSION_CONVERTER_POLICIES_FLC_12NOV2002_HPP
 #define BOOST_NUMERIC_CONVERSION_CONVERTER_POLICIES_FLC_12NOV2002_HPP
 
+#include <functional>
```
Comment (brandon-kohn) [inline include/boost/numeric/conversion/converter_policies.hpp:13]: Why was this moved?

File: include/boost/numeric/conversion/converter_policies.hpp [hunk 2]
```diff
@@ -137,22 +136,22 @@ class bad_numeric_cast : public std::bad_cast
 {
   public:
 
-    virtual const char * what() const BOOST_NOEXCEPT_OR_NOTHROW
+    const char * what() const BOOST_NOEXCEPT_OR_NOTHROW BOOST_OVERRIDE
```
Comment (brandon-kohn) [inline include/boost/numeric/conversion/converter_policies.hpp:139]: The BOOST_OVERRIDE is great. Removing the virtual keyword makes this method final to any types which may inherit. That breaks the interface. So I think the virtual should be maintained.

**Comment by EugeneZelenko** [comment]
Clang-tidy modernize-use-override warnings complains when both `virtual` and `override` are used. `virtual` is not required when virtual function is redefined, so `BOOST_OVERRIDE` is enough to understand that fact.
